### PR TITLE
Segmenting ISIC2018 with Improved Unet

### PR DESCRIPTION
**Pull request by valensmith**
This is a pull request to merge the recognition problem for Segmenting ISIC2018 with Improved Unet.

This pull request contains dataset.py, modules.py, train.py and predict.py

Software requirements can be found in the README.md

Cheers
Henry Smith - 45365737

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.85765. Predicted Mask output looks good.
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design.
- Sufficient comment in the code, but no docstring. Need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
